library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -286,149 +141,117 @@ public AbstractJdbcPollInputOperator()
+        || context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) > largestRecoveryWindow) {
+      scanService.scheduleAtFixedRate(new DBPoller(), 0, pollInterval, TimeUnit.MILLISECONDS);
+    }
+  }
+
+  protected void initializePreparedStatement()
+  {
+    try {
+      // If its a range query pass upper and lower bounds, If its a polling query pass only the lower bound
+      if (isPollerPartition) {
+        ps = store.getConnection().prepareStatement(buildRangeQuery(rangeQueryPair.getKey(), Integer.MAX_VALUE),
+            TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);
       } else {
         ps = store.getConnection().prepareStatement(
-            JdbcMetaDataUtility.buildPollableQuery(getTableName(), getKey(), rangeQueryPair.getKey()),
-            java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
-        isPollable = true;
+            buildRangeQuery(rangeQueryPair.getKey(), (rangeQueryPair.getValue() - rangeQueryPair.getKey())),
+            TYPE_FORWARD_ONLY, CONCUR_READ_ONLY);
       }
-
     } catch (SQLException e) {
       LOG.error("Exception in initializing the range query for a given partition", e);
       throw new RuntimeException(e);
     }
 
-    windowManager.setup(context);
-    LOG.debug("super setup done...");
   }
 
   @Override
   public void beginWindow(long windowId)
   {
     currentWindowId = windowId;
-
-    isReplayed = false;
-
     if (currentWindowId <= windowManager.getLargestRecoveryWindow()) {
       try {
         replay(currentWindowId);
+        return;
       } catch (SQLException e) {
         LOG.error("Exception in replayed windows", e);
         throw new RuntimeException(e);
       }
     }
-
-    if (isReplayed && currentWindowId == windowManager.getLargestRecoveryWindow()) {
-      try {
-        if (!isPollable && rangeQueryPair.getValue() != null) {
-
-          ps = store.getConnection().prepareStatement(
-              JdbcMetaDataUtility.buildGTRangeQuery(getTableName(), getKey(), previousUpperBound,
-                  rangeQueryPair.getValue()),
-              java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
-        } else {
-          String bound = null;
-          if (previousUpperBound == null) {
-            bound = getRangeQueryPair().getKey();
-          } else {
-            bound = previousUpperBound;
-          }
-          ps = store.getConnection().prepareStatement(
-              JdbcMetaDataUtility.buildPollableQuery(getTableName(), getKey(), bound),
-              java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
-          isPollable = true;
-        }
-        isReplayed = false;
-        LOG.debug("Prepared statement after re-initialization - {} ", ps.toString());
-      } catch (SQLException e) {
-        // TODO Auto-generated catch block
-        throw new RuntimeException(e);
-      }
+    if (isPollerPartition) {
+      updatePollQuery();
+      isPolled = false;
```

Comment:
non-poller partition should run only once. We make it false for poller partition because, it should try to poll again in next window to find incremental changes.